iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -1147,6 +1155,20 @@ class SchedulingBarrier(CustomOp):
     operations: list[Operation]
 
 
+@define_op("atomic_min")
+@dataclass
+class AtomicMin(BinaryOpBase, ABC):
+    """
+    Represents an atomic operation in the graph.
+    Takes in value and buffer to perform the operation on.
```

Comment:
Can you also specify what it returns because I think there's some confusion there about whether it returns the original value or the updated value